cmake/modules/RootConfiguration.cmake

```diff
@@ -560,9 +560,13 @@ if(PYTHON_VERSION_STRING_Development_Other)
 endif()
 
 #---RConfigure.h---------------------------------------------------------------------------------------------
-try_compile(has__cplusplus "${CMAKE_BINARY_DIR}" SOURCES "${CMAKE_SOURCE_DIR}/config/__cplusplus.cxx"
+if (CMAKE_CXX_COMPILER_ID STREQUAL "NVHPC")
+   execute_process(COMMAND ${CMAKE_CXX_COMPILER} -dM -E /dev/null OUTPUT_VARIABLE __cplusplus_PPout)
```

Comment:
Why is `try_compile` not working for nvc++

Reply:
Using `try_compile` would work as well:
```
   try_compile(has__cplusplus "${CMAKE_BINARY_DIR}" SOURCES "${CMAKE_SOURCE_DIR}/config/__cplusplus.cxx"
            COMPILE_DEFINITIONS -dM -E OUTPUT_VARIABLE __cplusplus_PPout)
```
But it would be irritating, since the source file is not actually read and compiled by the compiler. I cannot put `/dev/null` here, because that does not exist on Windows :) The importance is to compile with `-dM -E`, which asks the compiler to dump all predefined macros. The `pragma message` in `__cplusplus.cxx` is not supported by nvc++, so nothing is printed during compilation.

Reply:
Ah so semantically we probably need:
```
try_compile(has__cplusplus ...)
if ( has__cplusplus is empty )
   go the slower route
endif()
```
alternatively if it was actually resulted in a different code (but it likely is not)
```
has_pragma_message) = test (or aquire somehow) that compiler support pragma message
if (has_pragma_test)
    use try_compiler
else()
    use the slower route
endif()
```

i.e. this fix is not a priori depend of a specific compiler.